sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/receivesettle/sync/OrderSyncService.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus.receivesettle.sync;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.receivesettle.NetworkFailureException;
+import com.azure.messaging.servicebus.receivesettle.Order;
+import com.azure.messaging.servicebus.receivesettle.OrderServiceFailureException;
+
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+/**
+ * An order service that save orders and possibly throw exception for each operation.
+ */
+public class OrderSyncService {
+    private final Random rand = new Random();
+    private final ClientLogger logger = new ClientLogger(OrderSyncService.class);
+
+    /**
+     * Simulate an order creation or update
+     * Exception might be thrown before or after the order is saved (logged).
+     * @param order The order to be created or replaced.
+     */
+    public void createOrReplaceOrder(Order order) throws OrderServiceFailureException, NetworkFailureException {
+        this.throwRandomError();
+        if (order.getId() == null) {
+            Order newOrder = new Order();
+            newOrder.setId(UUID.randomUUID().toString());
+            newOrder.setContent(order.getContent());
+            // simulate creating a new order
+            logger.info(String.format("Order created: %s", newOrder));
+        } else {
+            // simulate updating the order into repository.
+            logger.info(String.format("Order updated %s", order));
+        }
+        this.throwRandomError();
```

Comment:
The sync operations should just call the async operation and block().

Reply:
I assume a user that uses sync API won't touch the async code. So I implement them separately.